test/test.py

```diff
@@ -1,133 +1,184 @@
 
     def test_munge_complex(self):
         """Tests munge.split_corpus and munge.write_clean_corpus on a real-world
         example corpus (Don Quixote). Uses same tests from test_split_corpus_simple
         and test_write_clean_corpus_simple."""  # do i need to repeat what I'm checking for? or is this sufficient?
 
-        # split_corpus tests
-        for doc in munge.split_corpus("test_files/quixote.txt"):
-            self.assertTrue(len(doc.split()) <= 500)
+        # corpus_to_documents tests
+        corpus = munge.import_corpus("test_files/quixote.txt")
+        for doc in munge.corpus_to_documents(corpus)[:-1]:
+            # all documents (except last) are between 250 and 500 words
             self.assertTrue(len(doc.split()) >= 250)
+            self.assertTrue(len(doc.split()) <= 500)
+            # all documents (except last) either end on punctuation or are 500 words. 5 characters back to accommodate
+            # for extra characters like quotes and parentheses
             self.assertTrue(
-                len(doc.split()) == 500 or doc[-1] in (".", "!", "?"))
+                len(doc.split()) == 500 or "." in doc[-5:] or "!" in doc[-5:] or "?" in doc[-5:])
 
         # write_clean_corpus tests
-        with open("test_files/munged_quixote.txt", "r") as inf:
-            for i, line in enumerate(inf):
+        with open("test_files/munged_quixote.txt", "r") as in_file:
+            for i, line in enumerate(in_file):
                 features = line.split("\t")
                 self.assertEqual(features[0], str(
-                    SAMPLE_METADATA["quixote"]["ids"][i]))
+                    self.sample_metadata["quixote"]["ids"][i]))
                 self.assertEqual(
-                    features[1], SAMPLE_METADATA["quixote"]["names"][i])
+                    features[1], self.sample_metadata["quixote"]["names"][i])
                 self.assertEqual(len(features), 3)
 
-    @classmethod
-    def tearDownClass(cls):
-        """Deletes temporary files created for class. Runs after all tests in class"""
-        test_file_list = (["test_files/simple_angel_100.txt", "test_files/munged_angel_50.txt",
-                           "test_files/simple_whale_100.txt", "test_files/munged_quixote.txt"])
-        for fil in test_file_list:
-            os.remove(fil)
+
+class TestMalletMethods(PreprocessingUnitTestClass):
+    """Test class for methods in mallet.py: make_topic_model"""
+
+    # def test_make_topic_model(self):
+    #     """Tests that mallet.make_topic_model creates an LdaMallet class object, and raises
+    #     an exception if the path to mallet is not found"""
+    #     # raise exception if MALLET_PATH is not defined and mallet is not in path
+    #     with self.assertRaises(RuntimeError):
+    #         corpus = munge.corpus_to_doc_tokens(
+    #             munge.import_corpus("test_files/simple_whale_100.txt"))
+    #         mallet.make_topic_model(corpus, 10)
+
+    def test_gib(self):
```

Comment:
I know you're still working on this part but this isn't a great function name, what does "gib" mean here